src/analysis/mod.rs

```diff
@@ -69,7 +71,47 @@ pub fn run(env: &mut Env) {
     }
     if !to_analyze.is_empty() {
         error!("Not analyzed {} objects due unfinished dependencies", to_analyze.len());
+        return;
     }
+
+    // Analyze free functions as the last step once all types are analyzed
+    analyze_global_functions(env);
+}
+
+fn analyze_global_functions(env: &mut Env) {
+    let ns = env.library.namespace(library::MAIN_NAMESPACE);
+
+    let full_name = format!("{}.*", ns.name);
+
+    let obj = match env.config.objects.get(&*full_name) {
+        Some(obj) if obj.status.need_generate() => obj,
+        _ => return,
+    };
+
+    let functions: Vec<_> = ns.functions.iter().filter(|f| f.kind == library::FunctionKind::Global).collect();
+    if functions.is_empty() {
+        return;
+    }
+
+    let mut imports = imports::Imports::new();
+    imports.add("glib::translate::*", None);
+    imports.add("ffi", None);
+
+    let functions = functions::analyze(env, &functions, TypeId::tid_none(), &obj, &mut imports, None, None);
+
+    imports.clean_glib(env);
+
+    env.analysis.global_functions = Some(info_base::InfoBase {
+        full_name: full_name,
+        type_id: TypeId::tid_none(),
+        name: "*".into(),
+        functions: functions,
+        specials: Default::default(),
+        imports: imports,
+        version: None,
+        deprecated_version: None,
+        cfg_condition: None,
+    });
```

Comment:
InfoBase implement Default so don't need many default values
`type_id: TypeId::tid_none()` better stays to be same as `analyze` parameter